src/libappimage/utils/ElfFile.cpp

```diff
@@ -0,0 +1,151 @@
+// system
+extern "C" {
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+}
+
+// local
+#include "light_byteswap.h"
+#include "ElfFile.h"
+
+
+typedef Elf32_Nhdr Elf_Nhdr;
+
+
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#define ELFDATANATIVE ELFDATA2LSB
+#elif __BYTE_ORDER == __BIG_ENDIAN
+#define ELFDATANATIVE ELFDATA2MSB
+#else
+#error "Unknown machine endian"
+#endif
+
+namespace appimage {
+    namespace utils {
+        ElfFile::ElfFile(const std::string& path) : path(path), fname(path.c_str()) {}
+
+        uint16_t ElfFile::file16_to_cpu(uint16_t val) {
+            if (ehdr.e_ident[EI_DATA] != ELFDATANATIVE)
```

Comment:
Redundancy in the next three methods. This could be `static (inline)`.

Reply:
Don't see the redundancy here. They are like safe invocations of `bswap_*` methods.
`static`: fits in.
`inline`: the compiler may ignore it not big gain here.

Reply:
It's this line:

```cpp
if (ehdr.e_ident[EI_DATA] != ELFDATANATIVE)
```

This check should be refactored out into a separate method.